src/OpenTelemetry.ResourceDetectors.Container/ContainerResourceDetector.cs

```diff
@@ -132,45 +141,107 @@ private static string RemovePrefixAndSuffixIfNeeded(string input, int startIndex
         return input.Substring(startIndex, endIndex - startIndex);
     }
 
+    private string? ExtractK8sContainerId()
+    {
+        try
+        {
+            var baseUrl = this.k8sMetadataFetcher.GetServiceBaseUrl();
+            if (string.IsNullOrEmpty(baseUrl))
+            {
+                return null;
+            }
+
+            var @namespace = this.k8sMetadataFetcher.GetNamespace();
+            var hostname = this.k8sMetadataFetcher.GetPodName() ?? this.k8sMetadataFetcher.GetHostname();
+            var containerName = this.k8sMetadataFetcher.GetContainerName();
```

Comment:
ContainerName is only parameter that should be provided by user manually. Should we check it first and exit if it was not provided? If it is provided, most probably user would like detector to resolve container ID through K8s API.
I thought whether we would like to have errors in log if token/namespace/certificate file are not available or should we check at least one with File.Exists before. Probably current implementation would be good, as we already would check some existence of environment variable and could reasonable think that we identified K8s environment.

Reply:
> If it is provided, most probably user would like detector to resolve container ID through K8s API.

I'm not sure to understand this part. Do you mean that if the user doesn't provide the container name, it means he wants to resolve the container ID via CGroup?

Reply:
If user have not provided  container name, we will not be able to resolve it with K8s API. We have several options to treat it:
- (1.a) probably we are not on K8s environment and we should not do anything
- (1.b) probably user don't want us to resolve container id with K8s API, for example he also have not configured rights
- (2) we make sure that we are in K8s environment b other means (Kubernetes server URL environment variable + expected files exist) and we believe it is user error that container name was not provided to us. In that case we should still special handle it with proper error message and advise to look in README.

It is up to you, which option (1 or 2) you like more.

Reply:
In option 2, we still should not do HTTP request, as it take time but we still will not be able to resolve container id.

Reply:
I did something: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1699/commits/2977bc7e66919e4c7652a2a60015f7338b5c3f9d. Please tell me what you think.